wpilibj/src/main/java/edu/wpi/first/wpilibj/experimental/command/SelectCommand.java

```diff
@@ -0,0 +1,99 @@
+package edu.wpi.first.wpilibj.experimental.command;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Supplier;
+
+import edu.wpi.first.wpilibj.command.IllegalUseOfCommandException;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Runs one of a selection of commands, either using a selector and a key->command mapping, or a
+ * supplier that returns the command directly at runtime.  Does not actually schedule the selected
+ * command - rather, the command is run through this command; this ensures that the command will
+ * behave as expected if used as part of a CommandGroup.  Requires the requirements of all included
+ * commands, again to ensure proper functioning when used in a CommandGroup.  If this is undesired,
+ * consider using {@link ScheduleCommand}.
+ *
+ * <p>As this command contains multiple component commands within it, it is technically a command
+ * group; the command instances that are passed to it cannot be added to any other groups, or
+ * scheduled individually.
+ *
+ * <p>As a rule, CommandGroups require the union of the requirements of their component commands.
+ */
+public class SelectCommand extends SendableCommandBase {
+
+  private final Map<Object, Command> m_commands;
+  private final Supplier<Object> m_selector;
+  private final Supplier<Command> m_toRun;
+  private Command m_selectedCommand;
+
+  /**
+   * Creates a new selectcommand.
+   *
+   * @param commands the map of commands to choose from
+   * @param selector the selector to determine which command to run
+   */
+  public SelectCommand(Map<Object, Command> commands, Supplier<Object> selector) {
+    if (!Collections.disjoint(CommandGroupBase.getGroupedCommands(), commands.values())) {
+      throw new IllegalUseOfCommandException("Commands cannot be added to multiple CommandGroups");
+    }
+
+    CommandGroupBase.registerGroupedCommands(commands.values().toArray(new Command[]{}));
+
+    requireNonNull(commands);
```

Comment:
Provide a message here and on the next line. You could also combine the check and assignments like:

```java
m_commands = requireNonNull(commands, "message");
```